src/util/dbid.cpp

```diff
@@ -0,0 +1,16 @@
+#include "util/dbid.h"
+
+
+const QVariant::Type DbId::kVariantType = DbId().toVariant().type();
+
+//static
+DbId::value_type DbId::valueOf(QVariant /*pass-by-value*/ variant) {
```

Comment:
Otherwise when passing by const-ref we would need to make a copy of the 
parameter anyway, because the implementation needs to call 
QVariant::convert(). Passing by value in the first place may allow the 
compiler to apply some optimizations.

On 08/24/2015 12:19 PM, Daniel Schürmann wrote:
>
> In src/util/dbid.cpp 
> <https://github.com/mixxxdj/mixxx/pull/684#discussion_r37737407>:
>
> > @@ -0,0 +1,16 @@
> > +#include "util/dbid.h"
> > +
> > +
> > +const QVariant::Type DbId::kVariantType = DbId().toVariant().type();
> > +
> > +//static
> > +DbId::value_type DbId::valueOf(QVariant /*pass-by-value*/ variant) {
>
> Why it is required to pass by value?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/684/files#r37737407>.
>